src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructure.java

```diff
@@ -0,0 +1,286 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * This class computes a number of useful structural properties for DAG specified by the
+ * immediate parents and immediate children of each node.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class ComputableGraphStructure {
+
+    private final Set<String> nodeKeysSet;
+    private final Set<String> nodeTagsSet;
```

Comment:
I wrote a description for tags in the javadoc for `ImmutableComputableGraph` and put a reference in the javadoc of `ComputableGraphStructure` to that description.